libass/ass_directwrite.c

```diff
@@ -426,6 +476,43 @@ static int encode_utf16(wchar_t *chars, uint32_t codepoint)
     }
 }
 
+static char *get_utf8_name(IDWriteLocalizedStrings *names, int k)
+{
+    wchar_t *temp_name = NULL;
+    char *mbName = NULL;
+
+    UINT32 length;
+    HRESULT hr = IDWriteLocalizedStrings_GetStringLength(names, k, &length);
+    if (FAILED(hr))
+        goto cleanup;
+
+    if (length >= (UINT32) -1 || length > SIZE_MAX / sizeof(wchar_t) - 1)
+        goto cleanup;
+
```

Comment:
Regarding the warning: Iirc, local suppresion of warnings is compiler dependent and would likely require a fair bit of `#ifdef`-checks. But I think just replacing `>` with `>=` will also get rid of the warning *(at the expense of lowering the theoretical limit by 1, which is probably ok)*.

Reply:
> just replacing `>` with `>=` will also get rid of the warning

Will it? When the warning occurs, `SIZE_MAX` is roughly 2^64, but `length` is at most 2^32.

I’m inclined to think we’ll just have to live with the warning.

Reply:
Nope, I brainfarted, sorry for the noise.
But moving the `- 1` offset to the left like so will – at least with gcc 10 and clang 12:
```C
(length + (size_t)1) > SIZE_MAX / sizeof(wchar_t)
```
Maybe the warning will reoccur with future compiler versions though

Reply:
Oh, I like that.

It’s because it sees `size_t` instead of `UINT32`, so just casting `length` would work, too. But your suggestion is cleaner as it doesn’t need to rely on the fact that `length` fits it `size_t`. And `length + 1` can’t overflow because of the check just before. Nice.